File: VB/SpreadsheetChartAPISamples/CodeExamples/DataTableActions.vb
Imports DevExpress.Spreadsheet
Imports DevExpress.Spreadsheet.Charts

Namespace SpreadsheetChartAPIActions
    Module DataTableActions
        Private Sub ShowDataTables(ByVal workbook As IWorkbook)
#Region "#ShowDataTable"
            Dim worksheet As Worksheet = workbook.Worksheets("chartTask5")
            workbook.Worksheets.ActiveWorksheet = worksheet

            'Create a chart and specify its location
            Dim chart As Chart = worksheet.Charts.Add(ChartType.Line, worksheet("B2:C8"))
            chart.TopLeftCell = worksheet.Cells("F2")
            chart.BottomRightCell = worksheet.Cells("L14")
            Dim dataTableOptions As DataTableOptions = chart.DataTable
            dataTableOptions.Visible = True
            dataTableOptions.ShowLegendKeys = False
#End Region
        End Sub

        Private Sub ChangeDataTableBorders(ByVal workbook As IWorkbook)
#Region "#ChangeDataTableBorders"
            Dim worksheet As Worksheet = workbook.Worksheets("chartTask5")
            workbook.Worksheets.ActiveWorksheet = worksheet

            'Create a chart and specify its location
            Dim chart As Chart = worksheet.Charts.Add(ChartType.Line, worksheet("B2:C8"))
            chart.TopLeftCell = worksheet.Cells("F2")
            chart.BottomRightCell = worksheet.Cells("L14")
            Dim dataTableOptions As DataTableOptions = chart.DataTable
            dataTableOptions.Visible = True
            dataTableOptions.ShowLegendKeys = False

            dataTableOptions.ShowVerticalBorder = False
            dataTableOptions.ShowOutline = False
#End Region
        End Sub

        Private Sub ChangeDataTableFont(ByVal workbook As IWorkbook)
#Region "#ChangeDataTableFont"
            Dim worksheet As Worksheet = workbook.Worksheets("chartTask5")
            workbook.Worksheets.ActiveWorksheet = worksheet

            'Create a chart and specify its location
            Dim chart As Chart = worksheet.Charts.Add(ChartType.Line, worksheet("B2:C8"))
            chart.TopLeftCell = worksheet.Cells("F2")
            chart.BottomRightCell = worksheet.Cells("L14")
            Dim dataTableOptions As DataTableOptions = chart.DataTable
            dataTableOptions.Visible = True
            dataTableOptions.ShowLegendKeys = False

            dataTableOptions.Font.Name = "Helvetica"
            dataTableOptions.Font.Size = 12
#End Region
        End Sub
    End Module
End Namespace

File: CS/SpreadsheetChartAPISamples/CodeExamples/DataTableActions.cs
using DevExpress.Spreadsheet;
using DevExpress.Spreadsheet.Charts;

namespace SpreadsheetChartAPIActions
{
    public static class DataTableActions
    {
        static void ShowDataTables(IWorkbook workbook)
        {
            #region #ShowDataTable
            Worksheet worksheet = workbook.Worksheets["chartTask5"];
            workbook.Worksheets.ActiveWorksheet = worksheet;

            // Create a chart and specify its location
            Chart chart = worksheet.Charts.Add(ChartType.Line, worksheet["B2:C8"]);
            chart.TopLeftCell = worksheet.Cells["F2"];
            chart.BottomRightCell = worksheet.Cells["L14"];
            DataTableOptions dataTableOptions = chart.DataTable;
            dataTableOptions.Visible = true;
            dataTableOptions.ShowLegendKeys = false;

            #endregion #ShowDataTable
        }

        static void ChangeDataTableBorders(IWorkbook workbook) 
        {
            #region #ChangeDataTableBorders
            Worksheet worksheet = workbook.Worksheets["chartTask5"];
            workbook.Worksheets.ActiveWorksheet = worksheet;

            // Create a chart and specify its location
            Chart chart = worksheet.Charts.Add(ChartType.Line, worksheet["B2:C8"]);
            chart.TopLeftCell = worksheet.Cells["F2"];
            chart.BottomRightCell = worksheet.Cells["L14"];
            DataTableOptions dataTableOptions = chart.DataTable;
            dataTableOptions.Visible = true;
            dataTableOptions.ShowLegendKeys = false;

            dataTableOptions.ShowVerticalBorder = false;
            dataTableOptions.ShowHorizontalBorder = false;
            #endregion #ChangeDataTableBorders
        }

        static void ChangeDataTableFont(IWorkbook workbook)
        {
            #region #ChangeDataTableFont
            Worksheet worksheet = workbook.Worksheets["chartTask5"];
            workbook.Worksheets.ActiveWorksheet = worksheet;

            // Create a chart and specify its location
            Chart chart = worksheet.Charts.Add(ChartType.Line, worksheet["B2:C8"]);
            chart.TopLeftCell = worksheet.Cells["F2"];
            chart.BottomRightCell = worksheet.Cells["L14"];
            DataTableOptions dataTableOptions = chart.DataTable;
            dataTableOptions.Visible = true;
            dataTableOptions.ShowLegendKeys = false;

            dataTableOptions.Font.Name = "Helvetica";
            dataTableOptions.Font.Size = 12;
            #endregion #ChangeDataTableFont
        }

    }
}
